arch/arm/mcount-insn.c

```diff
@@ -0,0 +1,146 @@
+#include "libmcount/internal.h"
+#include "mcount-arch.h"
+
+#define INSN_SIZE  8
+
+#ifdef HAVE_LIBCAPSTONE
+#include <capstone/capstone.h>
+#include <capstone/platform.h>
+
+void mcount_disasm_init(struct mcount_disasm_engine *disasm)
+{
+	if (cs_open(CS_ARCH_ARM, CS_MODE_ARM, &disasm->engine) != CS_ERR_OK) {
+		pr_dbg("failed to init Capstone disasm engine\n");
+		return;
+	}
+
+	if (cs_option(disasm->engine, CS_OPT_DETAIL, CS_OPT_ON) != CS_ERR_OK)
+		pr_dbg("failed to set detail option\n");
+}
+
+void mcount_disasm_finish(struct mcount_disasm_engine *disasm)
+{
+	cs_close(&disasm->engine);
+}
+
+/* return 0 if it's ok, -1 if not supported, 1 if modifiable */
+static int check_prologue(struct mcount_disasm_engine *disasm, cs_insn *insn)
+{
+	int i;
+	cs_arm *arm;
+	cs_detail *detail;
+	bool branch = false;
+	int status = -1;
+
+	/*
+	 * 'detail' can be NULL on "data" instruction
+	 * if SKIPDATA option is turned ON
+	 */
+	if (insn->detail == NULL)
+		return -1;
+
+	/* try to fix some PC-relative instructions */
+	if (insn->id == ARM_INS_ADR)
+		return 1;
```

Comment:
Please remove this until you actually handle ADR instruction.

Reply:
Okay. I will.